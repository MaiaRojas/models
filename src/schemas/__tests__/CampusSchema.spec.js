const mongoose = require('mongoose/browser');
const { CampusSchema } = require('../')(mongoose);

describe('CampusSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, CampusSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate document with valid data', () => {
    const doc = new mongoose.Document({
      slug: 'lim',
      name: 'Lima',
      locale: 'es-PE',
      timezone: 'America/Lima',
      active: true,
    }, CampusSchema);

    return doc.validate();
  });
});
