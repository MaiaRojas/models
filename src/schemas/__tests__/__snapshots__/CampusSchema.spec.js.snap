// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampusSchema should fail validation when fields missing 1`] = `
Object {
  "locale": [ValidatorError: Path \`locale\` is required.],
  "name": [ValidatorError: Path \`name\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
  "timezone": [ValidatorError: Path \`timezone\` is required.],
}
`;
